Find tasks by tag

kaixiangtay opened the pull request.

kaixiangtay committed:
Merge commit '6fc7e1a23dea70d87dfbf282ded6cf3de8cdb85f'

* commit '6fc7e1a23dea70d87dfbf282ded6cf3de8cdb85f':
  Minor changes to UG
  Update DG
  Update UG
  Update DG with use cases
  edit DeveloperGuide.md
  edit UserGuide.md
  Update UG and DG by Han Bin
  Update DG with 3 use cases
  Update Non-functional requirements in DeveloperGuide.md
  Update find task command section in UserGuide.md

kaixiangtay committed:
Merge commit '84535c1d06abd4a7721e53b071f5f7b48e375c92'

* commit '84535c1d06abd4a7721e53b071f5f7b48e375c92':
  Update DG glossary

kaixiangtay committed:
Merge commit '70016c4ccb4817a199f1ac62cfdf3404a66d9cc8'

* commit '70016c4ccb4817a199f1ac62cfdf3404a66d9cc8':
  Update Config file
  Update UG for add command format
  Make Task fields optional except Title
  Fix checkstyle errors
  Refactor Name field into Title
  Refactor all occurrences of Person into Task
  Try fixing CI

kaixiangtay committed:
Create TagContainsKeyword.java and rename TitleContainsKeywordsPredicate.java to DescriptionContainsKeywordsPredicate.java

kaixiangtay committed:
Add TagContainsKeywordsPredicateTest.java and fix checkstyle errors

kaixiangtay committed:
Handle empty tag input into find command and add new test cases to TagContainsKeyWordsPredicateTest

kaixiangtay committed:
Add extra test cases into FindCommandParserTest.java

Run through test coverage report, found out
TagContainsKeywordsPredicate in parse function and
handleSearchByTag function in FindCommandParser
are not covered in testing.

kaixiangtay committed:
Update UserGuide by adding a section for find a task by tag

habi39 reviewed (approved):
GJ, will merge conflicts

kaixiangtay committed:
Merge branch 'master' into findTasksByTag

kaixiangtay committed:
Merge commit '68c908c354c7a3f5a39e078ba14afc82ef63f4ca'

* commit '68c908c354c7a3f5a39e078ba14afc82ef63f4ca':
  Edit Add and Find command usage message
  Format TextArea and Separator for descriptions
  Add multiline description, line wrapping in input
  Change some planner to planit
  Show apprpriate tasks in typicalTasks and SampleData
  Update UML diagrams with Description and Planner
  Fix checkstyle error
  Change Address to Description
  Update DG addressbook references with planner
  Fix checkstyle errors
  Change Address Book into Planner

kaixiangtay committed:
Resolve conflicts in FindCommandParserTest.java

# Conflicts:
#	src/test/java/seedu/address/logic/parser/FindCommandParserTest.java

kaixiangtay committed:
Merge branch 'findTasksByTag' of https://github.com/kaixiangtay/tp into findTasksByTag

* 'findTasksByTag' of https://github.com/kaixiangtay/tp:
